app/lib/tutorial.cjsx

```diff
@@ -53,55 +53,49 @@ module.exports = React.createClass
             mediaByID
 
         awaitTutorialMedia.then (mediaByID) =>
-          Dialog.alert(<TutorialComponent steps={tutorial.steps} media={mediaByID} />, {
+          Dialog.alert(<TutorialComponent tutorial={tutorial} media={mediaByID} preferences={preferences} user={user} geordi={geordi} />, {
             className: 'tutorial-dialog',
             required: true,
             closeButton: true
           })
             .catch =>
               null # We don't really care if the user canceled or completed the tutorial.
-            .then =>
-              @markComplete tutorial, user
-
-    markComplete: (tutorial, user) ->
-      now = new Date().toISOString()
-      completedThisSession[tutorial.id] = now
-
-      if user?
-        tutorial.get('project').then (project) ->
-          user.get('project_preferences', project_id: project.id).then ([projectPreferences]) ->
-            projectPreferences ?= apiClient.type('project_preferences').create
-              links:
-                project: project.id
-              preferences: {}
-            # Build this manually. Having an index (even as a strings) keys creates an array.
-            projectPreferences.preferences ?= {}
-            projectPreferences.preferences.tutorials_completed_at ?= {}
-            projectPreferences.update "preferences.tutorials_completed_at.#{tutorial.id}": now
-            projectPreferences.save()
 
   propTypes:
-    steps: React.PropTypes.arrayOf React.PropTypes.shape
-      media: React.PropTypes.string
-      content: React.PropTypes.string
+    geordi: React.PropTypes.object
+    preferences: React.PropTypes.shape
+      preferences: React.PropTypes.object
+    tutorial: React.PropTypes.shape
+      steps: React.PropTypes.arrayOf React.PropTypes.shape
+        media: React.PropTypes.string
+        content: React.PropTypes.string
+    user: React.PropTypes.object
 
   getDefaultProps: ->
-    steps: []
+    geordi: {}
     media: {}
+    preferences: null
+    tutorial: {}
+    user: null
 
-  render: ->
+  componentDidMount: ->
+    @props.geordi.makeHandler 'tutorial-completion'
```

Comment:
Same note as above. I'm not positive this is doing anything unless it's used as a click handler. Since you're passing geordi straight in as a prop and doing your logging later, I think this can be removed. I'd double check though.